pkg/catalog/endpoint.go

```diff
@@ -53,11 +54,11 @@ func (mc *MeshCatalog) ListAllowedEndpointsForService(downstreamIdentity identit
 		return nil, err
 	}
 
-	// allowedEndpoints comprises of only those endpoints from outboundEndpoints that matches the endpoints from listEndpointsforIdentity
+	// allowedEndpoints comprises of only those endpoints from outboundEndpoints that matches the endpoints from listEndpointsForServiceIdentity
```

Comment:
🤔 This might be the wrong abstraction... Thanks for challenging this! I'll dig into it.

Reply:
@michelleN I started by merely trying to change the type here so it does not reference `K8sServiceAccount` and uses `ServiceIdentity`.
I see that we have bigger problems with `ListAllowedEndpointsforService`.  I recorded these in:
  - **catalog: Simplify ListAllowedEndpointsForService** #3190

Would you agree it would be better to tackle simplifying this function in a separate PR (as part of #3190) - not to muddy the waters here?

Alternatively - I could revert the changes to `ListAllowedEndpointsForService` and keep it in its original form (by using the `.ToK8sServiceAccount()` and `.ToServiceIdentity()` temporary helpers), but not sure this would be any different.

I propose we work on this as a fast-follow to this PR here.